sme_finance_application_schema/test.py

```diff
@@ -21,22 +166,72 @@ def test_entity_json(self):
                 self.assertTrue(json.loads(content))
 
 
+def test_validity_of_data(data, schema_name):
+    with open('./sme_finance_application_schema/{}'.format(schema_name)) as f:
+        schema = json.loads(f.read())
+        validator = jsonschema.validators.Draft4Validator(schema)
+        patch_store(validator.resolver.store)
+        if not validator.is_valid(data):
+            raise Exception('invalid data according to {}: {}'.format(schema_name, validator.validate(SME_V5)))
+
+
+def test_completion_of_data(data, schema_name):
+    with open('./sme_finance_application_schema/{}'.format(schema_name)) as f:
+        schema = json.loads(f.read())
+        expected_fields = schema['properties'].keys()
+        for field in expected_fields:
```

Comment:
can you change this to be a test within the case please?

then we can do something along the lines of..
```py
for field in expected_fields:
    with self.subTest(schema=schema_name, field=field):
        self.assertContains(data, field)
```